### PR TITLE
Make PSD timeseries

### PR DESCRIPTION
**Pull request by micamu**
This is a code to plot the variation over time of the psd generated by the pipeline.
Sample plots for ER8A and ER8B are in
https://www.atlas.aei.uni-hannover.de/~miriam.cabero/LSC/ER8/psd_variation_er8a/
https://www.atlas.aei.uni-hannover.de/~miriam.cabero/LSC/ER8/psd_variation_er8b/
respectively.
Please have a look and send me comments.

### CODE REVIEW

**Comment by ahnitz** [comment]
Miriam, is this still a work in progress?

**Comment by micamu** [comment]
The code is ready for the ASD variation plots.
I also want to make plots of the amplitude relative to median, and add to the workflow.

**Comment by micamu** [comment]
How can assign this to Tito?

File: bin/hdfcoinc/pycbc_plot_psd_timefreq [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import pylab
+import pycbc.results
+from matplotlib.colors import LogNorm
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument('--psd-file', required=True,
+                    help='HDF file containing the PSDs.')
+parser.add_argument('--output-file', required=True,
+                    help='Name for the output plot.')
+parser.add_argument('--rel-median', choices=['True','False'], default=False,
```
Comment (titodalcanton) [inline bin/hdfcoinc/pycbc_plot_psd_timefreq]: You don't need to do the choices=[] thing here. This is a boolean option so you can use action='store_true'. Look for similar options for instance in pycbc_plot_singles_vs_params. Also, I think a better name for the option would be --normalize or --normalize-over-time.

File: bin/hdfcoinc/pycbc_plot_psd_timefreq [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                    help='HDF file containing the PSDs.')
+parser.add_argument('--output-file', required=True,
+                    help='Name for the output plot.')
+parser.add_argument('--rel-median', choices=['True','False'], default=False, 
+                    help='Select this option if you want to plot the amplitude '
+                         'relative to median (default %(default)s).')
+parser.add_argument('--bins', type=int, default=1000,
+                    help='Number of (logarithmic-spaced) frequency bins to plot'
+                         ' (default %(default)d).')
+parser.add_argument('--method', choices=['min','max','mean','median'],
```
Comment (titodalcanton) [inline bin/hdfcoinc/pycbc_plot_psd_timefreq]: I'd rename this option to --reduction-method or --freq-bin-reduction.

File: bin/hdfcoinc/pycbc_plot_psd_timefreq [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    cbLabel = 'Amplitude relative to median'
+else:
+    asdmin = min(psds[0][kmin:]) ** 0.5 * fac
+    asdmax = 1e-20
+    cbLabel = 'Amplitude Spectral Density (Strain / $\\sqrt{\\rm Hz}$)'
+ 
+for psd_segment in xrange(len(psds)):
+    logging.info('Plotting PSD segment number %d', psd_segment)
+    time = numpy.array( [ start[psd_segment] , end[psd_segment] ] )
+    if opts.rel_median == 'True':
```
Comment (titodalcanton) [inline bin/hdfcoinc/pycbc_plot_psd_timefreq]: Once you implement my note above, change this to just `if opts.normalize:`.

**Comment by micamu** [comment]
New commit implements Tito's comments and changes the ticklabes in the colorbar for the amplitude relative to median plot

**Comment by micamu** [comment]
The code is tested with the PSD obtained in O1 Analysis 3, all plots can be seen in https://www.atlas.aei.uni-hannover.de/~miriam.cabero/LSC/O1/analysis3_psdvariation/

**Comment by titodalcanton** [comment]
I'm merging this now. Will integrate into the workflow on a separate PR.

**Pull request merged**